### PR TITLE
Remove vars_files on play includes for upgrade playbooks.

### PR DESCRIPTION
**Pull request by abutcher**
Part of ansible v2 support for #686

### CODE REVIEW

**Comment by abutcher** [comment]
aos-ci-test

**Comment by abutcher** [comment]
3.2 HA upgrade failing in a strange spot.

```
TASK [Generate etcd backup] ****************************************************
fatal: [master5.example.com]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'etcd_data_dir'"}
fatal: [master6.example.com]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'etcd_data_dir'"}
fatal: [master4.example.com]: FAILED! => {"failed": true, "msg": "'dict object' has no attribute 'etcd_data_dir'"}
```

**Comment by detiber** [comment]
@abutcher indeed, I would have expeced the failure to take place in 'Check available disk space for etcd backup' prior to that failure, maybe that one is failing silently for some reason, though.

It does look like we are only setting etcd_data_dir if using embedded etcd or if /etc/etcd/etcd.conf exists, it could be that we aren't evaluating those cases completely right.

**Comment by sdodson** [comment]
aos-ci-test

**Comment by arilivigni** [comment]
aos-ci-test

**Comment by detiber** [comment]
aos-ci-test-extended

**Comment by detiber** [comment]
@abutcher from the test failure:
```
TASK: [fail ] ***************************************************************** 
failed: [10.8.171.97] => {"failed": true}
msg: Upgrade packages not found
failed: [10.8.171.93] => {"failed": true}
msg: Upgrade packages not found
failed: [10.8.171.95] => {"failed": true}
msg: Upgrade packages not found
```

**Comment by abutcher** [comment]
aos-ci-test-extended

**Comment by abutcher** [comment]
aos-ci-test-extended

**Comment by abutcher** [comment]
The `etcd_data_dir` issue boiled down to this:

```
fatal: [master5.example.com]: FAILED! => {"changed": false, "failed": true, "module_stderr": "", "module_stdout": "Traceback (most recent call last):\r\n  File \"/root/.ansible/tmp/ansible-tmp-1462921681.65-264437553045386/openshift_facts\", line 8029, in <module>\r\n    main()\r\n  File \"/root/.ansible/tmp/ansible-tmp-1462921681.65-264437553045386/openshift_facts\", line 2079, in main\r\n    protected_facts_to_overwrite)\r\n  File \"/root/.ansible/tmp/ansible-tmp-1462921681.65-264437553045386/openshift_facts\", line 1580, in __init__\r\n    protected_facts_to_overwrite)\r\n  File \"/root/.ansible/tmp/ansible-tmp-1462921681.65-264437553045386/openshift_facts\", line 1640, in generate_facts\r\n    facts = set_etcd_facts_if_unset(facts)\r\n  File \"/root/.ansible/tmp/ansible-tmp-1462921681.65-264437553045386/openshift_facts\", line 722, in set_etcd_facts_if_unset\r\n    ini_fp = StringIO.StringIO(ini_str)\r\nAttributeError: class StringIO has no attribute 'StringIO'\r\n", "msg": "MODULE FAILURE", "parsed": false}
```

Calling `StringIO()` fixes but this may need to be a conditional import?

**Comment by abutcher** [comment]
aos-ci-extended

**Comment by abutcher** [comment]
aos-ci-test-extended

**Comment by abutcher** [comment]
> Calling StringIO() fixes but this may need to be a conditional import?

Works as is.

**Comment by abutcher** [comment]
@detiber ptal

File: roles/openshift_facts/library/openshift_facts.py [hunk 1]
```diff
@@ -717,7 +717,8 @@ def set_etcd_facts_if_unset(facts):
         try:
             # Add a fake section for parsing:
             ini_str = '[root]\n' + open('/etc/etcd/etcd.conf', 'r').read()
-            ini_fp = StringIO.StringIO(ini_str)
+            # pylint: disable=not-callable
+            ini_fp = StringIO(ini_str)
```
Comment (detiber) [inline roles/openshift_facts/library/openshift_facts.py]: Does the import for StringIO need to change? https://docs.python.org/2/library/stringio.html

I suspect the work in Ansible to support python3 may be coming into play here as well.

**Comment by detiber** [comment]
I think I want to give this change a bit more soak time (and QE time)

Thoughts on updating the upgrade entrypoints to require ansible-1.9.4 to unblock the other PR for 2.0 support?

File: roles/openshift_facts/library/openshift_facts.py (hunk 1 above)
Comment (abutcher) [inline roles/openshift_facts/library/openshift_facts.py]: It probably needs to change. I had originally thought that the extended tests would cover testing but our CI clusters have integrated etcd.

**Comment by abutcher** [comment]
@detiber Sounds good.

**Comment by abutcher** [comment]
aos-ci-test-extended

**Comment by abutcher** [comment]
aos-ci-test-extended

**Comment by abutcher** [comment]
@detiber Updated to use `io.StringIO` which is compatible with both v2/v3 and I've tested upgrades with each version.

**Comment by abutcher** [comment]
aos-ci-test-extended

**Comment by abutcher** [comment]
Non containerized upgrade failure is a flake. If we've set the `etcd_data_dir` fact and backed up etcd then the `openshift_facts` changes work with 1.9.4.

**Comment by abutcher** [comment]
aos-ci-test-extended

**Pull request merged**